src/libpmemfile/preload.c

```diff
@@ -501,6 +480,2728 @@ init_hooking(void)
 	 * Install the callback to be calleb by the syscall intercepting library
 	 */
 	intercept_hook_point = &hook;
+
+#ifndef NDEBUG
+	/* XXX temporary -- checking the unified syscall filter table */
+	struct syscall_early_filter_entry f;
+
+	f = get_early_filter_entry(0);
+	assert(syscall_number_filter[0] == f.must_handle);
+	assert(syscall_needs_fd_rlock[0] == f.fd_rlock);
+	assert(syscall_needs_fd_wlock[0] == f.fd_wlock);
+	assert(syscall_has_fd_first_arg[0] == f.fd_first_arg);
+	assert(syscall_needs_pmem_cwd_rlock[0] == f.cwd_rlock);
```

Comment:
You've gotta be kidding... Almost 3 thousand lines of repeated code instead of 10 line loop?